### ISSUE TITLE
Simpletask doesn't ask for storage permission on Android 6

### ISSUE THREAD

**Issue by Heiko-L** [mpcjanssen/simpletask-android#340]
Hi!

After configuring my sd card as internal storage I cannot access my todo.txt anymore.

https://developer.android.com/about/versions/marshmallow/android-6.0.html#adoptable-storage

Is a fix possible?

TIA,
Heiko

**Comment by tasn** [comment]
For whatever its worth, I can't access my todo.txt at all since upgrading to 6. I also have an sd card, but mine is not set to adoptable.

**Comment by mpcjanssen** [comment]
@tasn it would be helpful to get some log output. I assume this is for cloudless. You can share the logging from the settings (at the bottom). You can send it to my personal email (mpc.janssen@gmail.com) although the logging should not contain any private info. 

@Heiko-L I read the link, but Simpletask doesn't use any of the methods that are impacted by adoptable storage so I am not sure what might be the cause. Can you also send a log?

**Comment by Heiko-L** [comment]
Sorry, it was my fault :(

With Android 6 one has to enable permissions after installation (the app should ask for it?). I had to manually permit calendar and storage permission. Now it works.
Maybe I configured Cyanogenmod 13 so that it disables all permissions by default and because of that apps don't ask for it.

https://support.google.com/googleplay/answer/6270602?hl=en

And yes, for me it is Simpletask Cloudless.

Thank you for your work!

**Comment by mpcjanssen** [comment]
@Heiko-L you are right, the app should ask for storage permission, I still need to add the code for this. I hope this will be in the next version.

**Comment by tasn** [comment]
Thanks to @Heiko-L I solved my problem.

As I was writing my reply, @mpcjanssen wrote exactly what I was about to say. The problem is that simpletask cloudless doesn't ask for the storage permission. It was off by default, and I was none the wiser.

**Comment by Heiko-L** [comment]
So I think this issue can be closed?

**Comment by mpcjanssen** [comment]
I renamed the issue, I will still need to fix it.

**Comment by eduncan911** [comment]
Confirming this is still a bug as of today with latest Play version.

ANDROID 6.0.1 Stock (non-root)
Simple task Cloudless

Aa mentioned above, manually enabling the permissions worked.  But, I had no idea to do that.  Was just about to uninstall it and move on, but though I'd try the GH issues. Glad I found this one.

Error at first was I was never able to select any file or folder.  I thought the built in file browser was broken.

Then when I force the name change to be something like the stated ./data/nl.mpcjanssen.simpletask/todo.txt, and updates my sync app to use this directory, then I got a "Permission Denied" when trying to open my todo.txt formatted existing file.

Yeah, it needs to ask for permissions...
